src/org/zaproxy/zap/extension/stats/StatsdClient.java

```diff
@@ -0,0 +1,294 @@
+ * 
+ * 
+ * ZAP Changes:
+ * 	Originally copied from https://github.com/etsy/statsd/blob/master/examples/StatsdClient.java
+ *  Added License from https://github.com/etsy/statsd/blob/master/LICENSE
+ * 
+ */
+package org.zaproxy.zap.extension.stats;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.StandardSocketOptions;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.channels.DatagramChannel;
+import java.util.Locale;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.apache.log4j.Logger;
+
+public class StatsdClient extends TimerTask {
+    private ByteBuffer sendBuffer;
+    private Timer flushTimer;
+    private boolean multi_metrics = false;
+
+	private static final Random RNG = new Random();
+	private static final Logger log = Logger.getLogger(StatsdClient.class.getName());
+
+	private final InetSocketAddress _address;
+	private final DatagramChannel _channel;
+
+	public StatsdClient(String host, int port) throws UnknownHostException, IOException {
+		this(InetAddress.getByName(host), port);
+	}
+
+	public StatsdClient(InetAddress host, int port) throws IOException {
+		_address = new InetSocketAddress(host, port);
+		_channel = DatagramChannel.open();
+		/* Put this in non-blocking mode so send does not block forever. */
+		_channel.configureBlocking(false);
+		/* Increase the size of the output buffer so that the size is larger than our buffer size. */
+        _channel.setOption(StandardSocketOptions.SO_SNDBUF, 4096);
+        setBufferSize((short) 1500);
+	}
+
+    protected void finalize() {
+            flush();
+    }
+
+    public synchronized void setBufferSize(short packetBufferSize) {
+        if(sendBuffer != null) {
+            flush();
+        }
+        sendBuffer = ByteBuffer.allocate(packetBufferSize);
+    }
+
+    public synchronized void enableMultiMetrics(boolean enable) {
+        multi_metrics = enable;
+    }
+
+    public synchronized boolean startFlushTimer(long period) {
+        if(flushTimer == null) {
+            // period is in msecs 
+            if(period <= 0) { period = 2000; }
+            flushTimer = new Timer();
+
+            // We pass this object in as the TimerTask (which calls run())
+            flushTimer.schedule((TimerTask)this, period, period);
+            return true;
+        }
+        return false;
+    }
+
+    public synchronized void stopFlushTimer() {
+        if(flushTimer != null) {
+            flushTimer.cancel();
+            flushTimer = null;
+        }
+    }
+
+    public void run() {     // used by Timer, we're a Runnable TimerTask
+        flush();
+    }
+
+	public boolean timing(String key, int value) {
+		return timing(key, value, 1.0);
+	}
+
+	public boolean timing(String key, int value, double sampleRate) {
+		return send(sampleRate, String.format(Locale.ENGLISH, "%s:%d|ms", key, value));
```

Comment:
I dont _think_ so. We just specify keys which I think its ok to hardcode, so theres nothing to translate, as far as I can tell.